### PR TITLE
[Feature:CourseMaterials] Course Materials New Tag

### LINKED ISSUE #5383: Highlight which course materials files are new/un-viewed
If there are many files / links in the course materials folder, it would be good to highlight which ones the user has clicked on to either download or view in a separate popout browser window.  SImilar to how we use white/blue color coding to display viewed/un-viewed threads & posts in the discussion forum.

This is related to #5237 -- but this should specifically show which items are un-viewed.   Clearing the notification of a new file is not necessarily the same as viewing the file.

### PR DESCRIPTION
**Pull request by cjreed121**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
Right now you are only able to see what course materials there are but may be unsure if there is a new course material added.
Closes #5383

### What is the new behavior?
A new tag will appear to the right of the course material if it has not been viewed. Viewing, downloading, or downloading a containing folder will mark as an access to the course material. If a course material has at least 1 access it is considered viewed by that user.

### CODE REVIEW

**Comment by HenrikLam** [comment]
Got the following frog robot error when I tried to insert a link to _https://www.google.com_ with the name _link to google_
> Oh no! Something irrecoverable has happened...
> 
> app\exceptions\OutputException (Code: 0) thrown in /usr/local/submitty/site/app/libraries/Output.php (Line 362) by:
>             throw new OutputException("{$e->getMessage()} in {$e->getFile()}:{$e->getLine()}");
> 
> 
> Message:
> Key "/var/local/submitty/courses/f21/sample/uploads/course_materials/" for array with keys "/var/local/submitty/courses/f21/sample/uploads/course_materials, /var/local/submitty/courses/f21/sample/uploads/course_materials/link-link+to+google" does not exist. in /usr/local/submitty/site/app/templates/course/CourseMaterials.twig:164
> 
> Stack Trace:
> #0 /usr/local/submitty/site/app/views/course/CourseMaterialsView.php(137): app\libraries\Output->renderTwigTemplate()
> #1 unknown file(unknown line): app\views\course\CourseMaterialsView->listCourseMaterials()
> #2 /usr/local/submitty/site/app/libraries/Output.php(224): call_user_func_array()
> #3 /usr/local/submitty/site/app/libraries/Output.php(193): app\libraries\Output->renderTemplate()
> #4 /usr/local/submitty/site/app/libraries/response/WebResponse.php(43): app\libraries\Output->renderOutput()
> #5 /usr/local/submitty/site/public/index.php(140): app\libraries\response\WebResponse->render()

File: site/app/views/course/CourseMaterialsView.php [hunk 1]
```diff
@@ -21,9 +22,11 @@ public function listCourseMaterials(array $course_materials_db) {
         $base_course_material_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), 'uploads', 'course_materials');
         $directories = [];
         $directory_priorities = [];
+        $seen = [];
 
         /** @var CourseMaterial $course_material */
         foreach ($course_materials_db as $course_material) {
+            //$seen[$course_material->getPath()] = $course_material->userHasViewed($this->core->getUser()->getId());
```
Comment (HenrikLam) [inline site/app/views/course/CourseMaterialsView.php:29]: Can this be removed?

**Review by HenrikLam — APPROVED** [review]
Looks good. Might be good to make the badge refresh on click rather than on next visit but could be done in another PR.

**Review by HenrikLam — COMMENTED** [review]
Looks good when I turned debug mode off, but I'm still getting the frog robot when debug mode is on. Might be an error on my end so I will do a vagrant destroy and up and test it again soon. Might be good to make the badge refresh on click rather than on next visit but could be done in another PR.

**Comment by HenrikLam** [comment]
Everything seems to work after I did a vagrant destroy and up, I probably broke something on my end before.

**Comment by cjreed121** [comment]
This has a failing course materials test. The fix for this is in #6907 and this PR should not be merged until that is merged.

File: migration/migrator/data/course_tables.sql [hunk 2]
```diff
@@ -1770,11 +1809,11 @@ ALTER TABLE ONLY public.course_materials_sections
 
 
 --
--- Name: course_materials_sections fk_section_id; Type: FK CONSTRAINT; Schema: public; Owner: -
```
Comment (bmcutler) [inline migration/migrator/data/course_tables.sql:1773]: wait, why are we deleting/replacing a a constraint for course_materials_sections?

File: migration/migrator/data/course_tables.sql [hunk 3]
```diff
@@ -1770,11 +1809,11 @@ ALTER TABLE ONLY public.course_materials_sections
...
+-- Name: course_materials_access fk_user_id; Type: FK CONSTRAINT; Schema: public; Owner: -
 --
 
-ALTER TABLE ONLY public.course_materials_sections
```
Comment (bmcutler) [inline migration/migrator/data/course_tables.sql:1776]: same as above

**Review by bmcutler — CHANGES_REQUESTED** [review]

**Comment by bmcutler** [comment]
@cjreed121 
Hi Chris, can you please resolve the merge conflicts?

**Review by bmcutler — APPROVED** [review]

**Pull request merged**